src/main/java/io/kestra/storage/minio/MinioStorage.java

```diff
@@ -450,6 +454,27 @@ private IOException reThrowMinioStorageException(String uri, MinioException e) {
         return new IOException(e);
     }
 
+    private Map<String, String> toStoredMetadata(Map<String, String> metadata) {
+        if (metadata == null) {
+            return null;
+        }
+        return metadata.entrySet().stream()
+            .map(entry -> Map.entry(entry.getKey().replaceAll("([A-Z])", "_$1").toLowerCase(), entry.getValue()))
```

Comment:
`replaceAll(regex, string)` can be costly as it will result in a regex being on each call. It would be better to create a static field for the `[A-Z]` regex (i.e., `Pattern.compile(regex)`) and then use `pattern.matcher(str).replaceAll(repl)`

Reply:
fixed